internal/cache/metadata/stat_cache_test.go

```diff
@@ -543,5 +543,51 @@ func (t *StatCacheTest) Test_ShouldEvictEntryOnFullCapacityIncludingFolderSize()
 	assert.Equal(t.T(), "2", entry2.Name)
 	assert.True(t.T(), hit3)
 	assert.Equal(t.T(), "3", entry3.Name)
+}
 
+func (t *StatCacheTest) Test_ShouldEvictAllEntriesWithPrefixFolder() {
+	localCache := lru.NewCache(uint64(10000))
+	t.statCache = metadata.NewStatCacheBucketView(localCache, "local_bucket")
+	folderEntry1 := &gcs.Folder{
+		Name:           "a",
+		Metageneration: 1,
+	}
+	objectEntry1 := &gcs.MinObject{Name: "a/b"}
+	objectEntry2 := &gcs.MinObject{Name: "a/b/c"}
+	objectEntry3 := &gcs.MinObject{Name: "d"}
+	folderEntry2 := &gcs.Folder{
+		Name:           "a/d",
+		Metageneration: 1,
+	}
+	folderEntry3 := &gcs.Folder{
+		Name:           "b",
+		Metageneration: 1,
+	}
+	t.statCache.InsertFolder(folderEntry1, expiration) //adds size of 220 and exceeds capacity
+	t.statCache.Insert(objectEntry1, expiration)       // adds size of 1428
+	t.statCache.Insert(objectEntry2, expiration)       // adds size of 1428
+	t.statCache.InsertFolder(folderEntry2, expiration) //adds size of 220 and exceeds capacity
+	t.statCache.InsertFolder(folderEntry3, expiration) //adds size of 220 and exceeds capacity
+	t.statCache.Insert(objectEntry3, expiration)       // adds size of 1428
+
+	t.statCache.InValidateCacheForEntriesWithGivenPrefix("a")
+
+	hit1, entry1 := t.statCache.LookUpFolder("a", someTime)
+	hit2, entry2 := t.statCache.LookUpFolder("a/b", someTime)
+	hit3, entry3 := t.statCache.LookUp("a/b/c", someTime)
+	hit4, entry4 := t.statCache.LookUpFolder("a/d", someTime)
+	hit5, entry5 := t.statCache.LookUpFolder("b", someTime)
+	hit6, entry6 := t.statCache.LookUp("d", someTime)
+	assert.False(t.T(), hit1)
```

Comment:
Ideally you should do assert after each lookup. Proceed to 2nd assertion only if first one fails.

Reply:
Done